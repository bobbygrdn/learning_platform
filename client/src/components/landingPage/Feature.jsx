import React from 'react';
import '../../styles/landingPage/Feature.css';

export default function Feature({ feature }) {
    return (
        <div className='feature' data-testid="feature">
            <img className="icon" src={feature.Image} alt={`Feature ${feature.key}`} />
            <p className="text" data-testid="text">{feature.description}</p>
        </div>
    )
}
