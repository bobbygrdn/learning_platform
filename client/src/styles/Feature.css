.feature {
    display: flex;
    flex-direction: column;
    align-items: center;
    pointer-events: none;
}

.icon {
    width: 7em;
    height: 7em;
}

.text {
    font-size: 1rem;
    text-align: center;
}