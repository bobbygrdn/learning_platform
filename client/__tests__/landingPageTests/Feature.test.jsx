import React from 'react';
import { render, screen } from '@testing-library/react';
import Feature from '../../src/components/landingPage/Feature';


const feature = {
    description: "Custom learning paths for different goals and skill levels",
    Image: `${process.env.PUBLIC_URL + "/resources/learningPath_icon.png"}`,
}

it('renders without crashing', () => {
    const { container } = render(<Feature feature={feature} />);
    expect(container).toMatchSnapshot();
})

it('should render the feature image with icon class name', () => {
    render(<Feature feature={feature} />);
    expect(screen.getByRole('img')).toMatchSnapshot();
})

it('should render the feature description with text class name', () => {
    render(<Feature feature={feature} />);
    expect(screen.getAllByTestId('text')).toHaveLength(1);
})
