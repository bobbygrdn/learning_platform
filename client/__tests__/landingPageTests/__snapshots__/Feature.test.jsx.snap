// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div>
  <div
    class="feature"
    data-testid="feature"
  >
    <img
      alt="Feature undefined"
      class="icon"
      src="undefined/resources/learningPath_icon.png"
    />
    <p
      class="text"
      data-testid="text"
    >
      Custom learning paths for different goals and skill levels
    </p>
  </div>
</div>
`;

exports[`should render the feature image with icon class name 1`] = `
<img
  alt="Feature undefined"
  class="icon"
  src="undefined/resources/learningPath_icon.png"
/>
`;
